homeassistant/components/ps4/__init__.py

```diff
@@ -158,9 +164,28 @@ def save_games(hass: HomeAssistantType, games: dict):
     except OSError as error:
         _LOGGER.error("Could not save game list, %s", error)
 
-    # Retry loading file
-    if games is None:
-        load_games(hass)
+
+def _reformat_data(hass: HomeAssistantType, games: dict) -> dict:
```

Comment:
Looks like `games` can be `None` both as argument and return value, or is that check not needed?

Reply:
Yes you are correct but after looking at the json helper, it seems to return '{}' instead of None. So we should always be passing a dict correct? 
```
def load_json(filename: str, default: Union[List, Dict, None] = None) \
        -> Union[List, Dict]:
    """Load JSON data from a file and return as dict or list.

    Defaults to returning empty dict if file is not found.
    """
    try:
        with open(filename, encoding='utf-8') as fdesc:
            return json.loads(fdesc.read())  # type: ignore
    except FileNotFoundError:
        # This is not a fatal error
        _LOGGER.debug('JSON file not found: %s', filename)
    except ValueError as error:
        _LOGGER.exception('Could not parse JSON content: %s', filename)
        raise HomeAssistantError(error)
    except OSError as error:
        _LOGGER.exception('JSON file reading failed: %s', filename)
        raise HomeAssistantError(error)
    return {} if default is None else default

```

Reply:
We would have to handle HomeAssistantError though right?

Reply:
Yes! It'll be a dict or an error. :+1:

Reply:
Technically the json could be something else than a dict, right? We should habdke that  if necessary, as close to loading the json as possible. Ie, make sure we convert to dict asap.

Reply:
How would we go about this? We cant predict how the data would be structured right?

Reply:
We could check for dict after loading and log error if not and default to empty dict as we do on home assistant error. We normally expect the data to be a dict from the json, right? So a list would indicate some file corruption or other problem.